2966.	[bug]		isc_print_vsnprintf() failed to check if there was
			space available in the buffer when adding a left
			justified character with a non zero width,
			(e.g. "%-1c"). [RT #22270]

--- bind/lib/isc/print.c
+++ bind/lib/isc/print.c
@@ -468,7 +468,7 @@ isc_print_vsnprintf(char *str, size_t si
 			if (width > 0) {
 				count += width;
 				width--;
-				if (left) {
+				if (left && size > 1) {
 					*str++ = c;
 					size--;
 				}
